src/test/kotlin/com/github/jensim/megamanipulator/actions/git/CloneOperatorTest.kt

```diff
@@ -0,0 +1,292 @@
+package com.github.jensim.megamanipulator.actions.git
+
+import com.github.jensim.megamanipulator.actions.NotificationsOperator
+import com.github.jensim.megamanipulator.actions.ProcessOperator
+import com.github.jensim.megamanipulator.actions.apply.ApplyOutput
+import com.github.jensim.megamanipulator.actions.git.clone.CloneOperator
+import com.github.jensim.megamanipulator.actions.localrepo.LocalRepoOperator
+import com.github.jensim.megamanipulator.actions.vcs.PrRouter
+import com.github.jensim.megamanipulator.actions.vcs.PullRequestWrapper
+import com.github.jensim.megamanipulator.files.FilesOperator
+import com.github.jensim.megamanipulator.settings.ProjectOperator
+import com.github.jensim.megamanipulator.ui.UiProtector
+import com.intellij.notification.NotificationType
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.junit5.MockKExtension
+import io.mockk.mockk
+import io.mockk.verify
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.runBlocking
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import java.io.File
+
+@ExtendWith(MockKExtension::class)
+class CloneOperatorTest {
+
+    @RelaxedMockK
+    private lateinit var filesOperator: FilesOperator
+
+    @MockK
+    private lateinit var projectOperator: ProjectOperator
+
+    @MockK
+    private lateinit var prRouter: PrRouter
+
+    @MockK
+    private lateinit var localRepoOperator: LocalRepoOperator
+
+    @MockK
+    private lateinit var processOperator: ProcessOperator
+
+    @MockK
+    private lateinit var notificationsOperator: NotificationsOperator
+
+    @MockK
+    private lateinit var uiProtector: UiProtector
+
+    @InjectMockKs
+    private lateinit var cloneOperator: CloneOperator
+
+    companion object {
+        private const val CLONING_TITLE_AND_MESSAGE = "Cloning repos"
+        private const val PROJECT = "mega-manipulator"
+        private const val BASE_REPO = "base_repo"
+    }
+
+    @Test
+    fun `clone with search requests`() = runBlocking {
+        // Given
+        val state = listOf<Pair<String, List<String>>>()
+        every {
+            uiProtector.mapConcurrentWithProgress<String, List<String>>(
+                title = CLONING_TITLE_AND_MESSAGE,
+                extraText1 = CLONING_TITLE_AND_MESSAGE,
+                extraText2 = any(),
+                data = any(),
+                mappingFunction = any()
+            )
+        } returns state
+
+        every { projectOperator.project.basePath } returns ".tmp/mega-manipulator-path"
+        every { filesOperator.refreshConf() } returns Unit
```

Comment:
FilesOperator is created with relaxed, mocking a unit-return does nothing extra here?

Reply:
It's right, i'll remove it